internal/manifest/store.go

```diff
@@ -23,35 +23,49 @@ func NewStore(content fleetcontrollers.ContentController) Store {
 }
 
 type contentStore struct {
-	sync.RWMutex
-
 	contentCache fleetcontrollers.ContentCache
 	content      fleetcontrollers.ContentClient
 }
 
 func (c *contentStore) Store(manifest *Manifest) (string, error) {
-	data, id, err := manifest.Content()
+	id, err := manifest.ID()
 	if err != nil {
 		return "", err
 	}
 
-	_, err = c.contentCache.Get(id)
-	if err == nil {
-		return id, nil
-	} else if !apierrors.IsNotFound(err) {
+	if _, err := c.contentCache.Get(id); err != nil && !apierrors.IsNotFound(err) {
 		return "", err
+	} else if err == nil {
+		return id, nil
+	}
+
+	return id, c.createContents(id, manifest)
+}
+
+func (c *contentStore) createContents(id string, manifest *Manifest) error {
+	data, err := manifest.Content()
+	if err != nil {
+		return err
+	}
+	digest, err := manifest.SHASum()
+	if err != nil {
+		return err
 	}
 
+	// Contents do not exist in the cluster
 	compressed, err := content.Gzip(data)
 	if err != nil {
-		return id, err
+		return err
 	}
 
 	_, err = c.content.Create(&fleet.Content{
 		ObjectMeta: metav1.ObjectMeta{
 			Name: id,
+			Annotations: map[string]string{
```

Comment:
why do we use an annotation here? would it make sense to use a new status field as you added for the `Bundle` ?

Reply:
It's used by Lookup for verifying the integrity of the payload.
Since Lookup only has access to `Content` objects, not `Bundles`, you mean to add a new field to the `Content` object, right? (in fact, I would add it to the `Spec`, not in the `Status`.
That's a fair point, I don't mind adding that as well. The more aggressive approach would be to completely get rid of the integrity check, which I honestly think it does not make much sense.

Reply:
yes I meant the `Content` resource. The integrity check does not make much sense to me either, I think it can be removed

Reply:
Replaced annotation with a dedicated spec field in `Content`, thanks!
If you feel strongly about the integrity check, please reopen this conversation.